services/motion/builtin/builtin.go

```diff
@@ -157,20 +159,27 @@ func (ms *builtIn) Move(
 	goalPose, _ := tf.(*referenceframe.PoseInFrame)
 
 	// the goal is to move the component to goalPose which is specified in coordinates of goalFrameName
-	output, err := motionplan.PlanMotion(ctx, ms.logger, goalPose, movingFrame, fsInputs, frameSys, worldState, constraints, extra)
+	steps, err := motionplan.PlanMotion(ctx, ms.logger, goalPose, movingFrame, fsInputs, frameSys, worldState, constraints, extra)
 	if err != nil {
 		return false, err
 	}
 
 	// move all the components
-	for _, step := range output {
+	for _, step := range steps {
 		// TODO(erh): what order? parallel?
 		for name, inputs := range step {
 			if len(inputs) == 0 {
 				continue
 			}
 			err := resources[name].GoToInputs(ctx, inputs)
 			if err != nil {
+				for _, step := range steps {
```

Comment:
I think theres probably better way to write this than nesting this far in, but the idea is right